FundTrack.BLL/Concrete/FinOpService.cs

```diff
@@ -184,29 +190,114 @@ public IEnumerable<FinOpViewModel> GetFinOpsByOrgAccount(int orgAccountId)
         {
             try
             {
-                var finOps = _unitOfWork.FinOpRepository.GetFinOpByOrgAccount(orgAccountId)
-                    .OrderByDescending(f => f.Id)
-                    .Select(f => new FinOpViewModel
-                    {
-                        Id = f.Id,
-                        CardFromId = f.AccFromId.GetValueOrDefault(0),
-                        CardToId = f.AccToId.GetValueOrDefault(0),
-                        Date = f.FinOpDate,
-                        Description = f.Description,
-                        Amount = f.Amount,
-                        TargetId = f.TargetId,
-                        Target = f.Target.TargetName,
-                        FinOpType = f.FinOpType,
-                        IsEditable = true,
-                        DonationId = f.DonationId
-                    });
-                return finOps;
+                var finOps = _unitOfWork.FinOpRepository.GetFinOpByOrgAccount(orgAccountId);
+                return InitializeFinOps(finOps);
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessLogicException(ErrorMessages.EmptyFinOpList, ex);
+            }
+        }
+
+        public IEnumerable<FinOpViewModel> GetFinOpByOrgAccountIdForPage(int orgAccountId, int currentPage, int itemsPerPage, int finOpType)
+        {
+            try
+            {
+                var finOps = _unitOfWork.FinOpRepository.GetFinOpByOrgAccountIdForPage(orgAccountId, currentPage, itemsPerPage);
+                if (finOpType == Constants.AnyFinOpType)
+                {
+                    return InitializeFinOps(finOps);
+                }
+                else
+                {
+                    finOps.Where(f => f.FinOpType == finOpType);
+                    return InitializeFinOps(finOps);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessLogicException(ErrorMessages.EmptyFinOpList, ex);
+            }
+        }
+
+        private IEnumerable<FinOpViewModel> InitializeFinOps(IQueryable<FinOp> finOps)
+        {
+            try
+            {
+                var finOpList = finOps.Select(f => new FinOpViewModel
+                {
+                    Id = f.Id,
+                    CardFromId = f.AccFromId.GetValueOrDefault(0),
+                    CardToId = f.AccToId.GetValueOrDefault(0),
+                    Date = f.FinOpDate,
+                    Description = f.Description,
+                    Amount = f.Amount,
+                    TargetId = f.TargetId,
+                    Target = f.Target.TargetName,
+                    FinOpType = f.FinOpType,
+                    DonationId = f.DonationId
+                }).ToList();
+                foreach (var f in finOpList)
+                {
+                    f.IsEditable = IsEditable(f.Date, f.CardFromId, f.CardToId);

```

Comment:
why not assigning IsEditable in Select( f => new  new FinOpViewModel { }) above?